homeassistant/components/motion_blinds/gateway.py

```diff
@@ -48,3 +54,79 @@ async def async_connect_gateway(self, host, key):
             self.gateway_device.protocol,
         )
         return True
+
+    def check_interface(self):
+        """Check if the current interface supports multicast."""
+        try:
+            return self.gateway_device.Check_gateway_multicast()
+        except socket.timeout:
+            return False
+
+    async def async_get_interfaces(self):
+        """Get list of interface to use."""
+        interfaces = [DEFAULT_INTERFACE, "0.0.0.0"]
+        enabled_interfaces = []
+        default_interface = DEFAULT_INTERFACE
+
+        adapters = await network.async_get_adapters(self._hass)
+        for adapter in adapters:
+            if ipv4s := adapter["ipv4"]:
+                ip4 = ipv4s[0]["address"]
+                interfaces.append(ip4)
+                if adapter["enabled"]:
+                    enabled_interfaces.append(ip4)
+                    if adapter["default"]:
+                        default_interface = ip4
+
+        if len(enabled_interfaces) == 1:
+            default_interface = enabled_interfaces[0]
+            interfaces.remove(default_interface)
+            interfaces.insert(0, default_interface)
+
+        if self._interface is not None:
+            interfaces.remove(self._interface)
+            interfaces.insert(0, self._interface)
+
+        return interfaces
+
+    async def async_check_interface(self, host, key):
+        """Connect to the Motion Gateway."""
+        self._interfaces = await self.async_get_interfaces()
+        for interface in self._interfaces:
+            _LOGGER.debug(
+                "Checking Motion Blinds interface '%s' with host %s", interface, host
+            )
+            # initialize multicast listener
+            check_multicast = AsyncMotionMulticast(interface=interface)
+            try:
+                await check_multicast.Start_listen()
+            except socket.gaierror:
+                continue
+
+            # trigger test multicast
+            self._gateway_device = MotionGateway(
+                ip=host, key=key, multicast=check_multicast
+            )
+            result = await self._hass.async_add_executor_job(self.check_interface)
+
+            # close multicast listener again
+            try:
+                check_multicast.Stop_listen()
```

Comment:
Can this do blocking I/O?

Reply:
No it does not do I/O, it only calls `listen_couroutine.close()` which shuts down the couroutine task that is listening for multicast messages (which is async itself).

Reply:
It does these things like socket.close():
https://github.com/starkillerOG/motion-blinds/blob/2d9b75355785c1869114de157ffeb2b50491feb5/motionblinds/async_motion_blinds.py#L125-L134
But I don't think that is I/O or blocking right?

Reply:
Can you confirm this is fine as is?
Otherwise I will adjust it in the upstream library and also for some other integrations using the same methods.

Reply:
It looks safe on the face

Reply:
@bdraco can you then approve/merge this PR?